### PR TITLE
Prevent duplication of authors when merging abstracts

### LINKED ISSUE #2143: Abstract merge: primary author appears twice
If the primary author of both abstracts is the same, they will show up twice in the final abstract.

### PR DESCRIPTION
**Pull request by mic4ael**
fixes #2143

### CODE REVIEW

File: indico/MaKaC/review.py [hunk 1]
```diff
@@ -2095,9 +2098,13 @@ def mergeInto(self,responsible,targetAbs,mergeAuthors=False,comments=""):
             #    if self.isPrimaryAuthor(auth):
             #        targetAbs.addPrimaryAuthor(newAuth)
             for auth in self.getPrimaryAuthorList():
+                if auth in targetAbs.getPrimaryAuthorList():
```
Comment (ThiefMaster) [inline indico/MaKaC/review.py]: `O(n²)`? ;) Why not create a set from that list outside the loop and check against that instead?

File: indico/MaKaC/review.py [hunk 2]
```diff
@@ -343,6 +343,9 @@ def clone(self, abstract):
     def isSpeaker(self):
         return self._abstract.isSpeaker(self)
 
+    def __cmp__(self, other):
```
Comment (ThiefMaster) [inline indico/MaKaC/review.py]: is this needed?

Comment (mic4ael) [inline indico/MaKaC/review.py]: As long as I want to use `in` operator i think it is

File: indico/MaKaC/review.py [hunk 3]
```diff
@@ -343,6 +343,12 @@ def clone(self, abstract):
     def isSpeaker(self):
         return self._abstract.isSpeaker(self)
 
+    def __eq__(self, other):
+        return self.getData() == other.getData()
+
```
Comment (ThiefMaster) [inline indico/MaKaC/review.py]:
```python
def __ne__(self, other):
    return not (self == other)
```

**Pull request merged**